astra/src/test/java/com/slack/astra/server/ManagerApiGrpcTest.java

```diff
@@ -452,29 +460,805 @@ public void shouldUpdatePartitionAssignments() {
+  }
+
+  @Test
+  public void shouldAutoAssign2NoPartitions() {
+    createEmptyDatasetGRPC("testDataset", "testOwner");
+    assertThatThrownBy(() -> updatePartitionAssignmentGRPC("testDataset", 10))
+        .isInstanceOfSatisfying(
+            io.grpc.StatusRuntimeException.class,
+            withGrpcStatusAndDescription(
+                Status.FAILED_PRECONDITION, "Needed 2 partitions with enough capacity, found 0"));
+  }
+
+  @Test
+  public void shouldAutoAssign3NotEnoughPartitions() {
+    createEmptyDatasetGRPC("testDataset", "testOwner");
+    createPartitions("1");
+
+    assertThatThrownBy(() -> updatePartitionAssignmentGRPC("testDataset", 10))
+        .isInstanceOfSatisfying(
+            io.grpc.StatusRuntimeException.class,
+            withGrpcStatusAndDescription(
+                Status.FAILED_PRECONDITION,
+                "Needed 2 partitions with enough capacity, found 1: [1]"));
+  }
+
+  @Test
+  public void shouldInvalidArgumentOnUpdatingWithBlankDataset() {
+    assertThatThrownBy(() -> updatePartitionAssignmentGRPC("", 10))
+        .isInstanceOfSatisfying(
+            io.grpc.StatusRuntimeException.class,
+            withGrpcStatusAndDescription(
+                Status.INVALID_ARGUMENT, "Dataset name must not be blank string"));
+  }
+
+  @Test
+  public void shouldUpdatePartitionAutoAssignmentsSingleDatasetMetadata() {
+    String datasetName = "testDataset";
+    String datasetOwner = "testOwner";
+
+    createPartitions("1", "2", "3", "4", "5");
+
+    Metadata.DatasetMetadata initialDatasetRequest =
+        createEmptyDatasetGRPC(datasetName, datasetOwner);
+    assertThat(initialDatasetRequest.getPartitionConfigsList().size()).isEqualTo(0);
+
+    // first assignment: partitionList empty, ThroughputBytes provided, requireDedicatedPartition =
+    // False
+    long nowMs = Instant.now().toEpochMilli();
+    long throughputBytes = 10;
+    ManagerApi.UpdatePartitionAssignmentRequest request =
+        ManagerApi.UpdatePartitionAssignmentRequest.newBuilder()
+            .setName(datasetName)
+            .setThroughputBytes(throughputBytes)
+            .build();
+    managerApiStub.updatePartitionAssignment(request);
+    await()
+        .until(
+            () ->
+                datasetMetadataStore.listSync().size() == 1
+                    && datasetMetadataStore.listSync().get(0).getPartitionConfigs().size() == 1);
+
+    assertThat(getDatasetMetadataGRPC(datasetName))
+        .has(throughputBytesOf(throughputBytes))
+        .has(partitionConfigsSizeOf(1))
+        .has(partitionsForIndexOf(0, List.of("1", "2")))
+        .has(latestPartitionConfigWithIndexOf(0, nowMs));
+
+    assertThat(getPartitionMetadata("1", "2")).have(sharedPartitionsWithCapacity(5));
+
+    // second assignment: partitionList empty, ThroughputBytes not provided,
+    // requireDedicatedPartition = True
+    nowMs = Instant.now().toEpochMilli();
+    throughputBytes = -1;
```

Comment:
It's simpler if to fail on the case if throughputBytes is not a valid number. So, let's fail this case. Update this test to set a value to be less than full partition.

We can add another line here for dedicated true and capacity is MAX_THROUGHPUT_BYTES

Reply:
The existing behavior is to consider -1 to mean keep the current throughput bytes setting.